lib/multiplexer/multiplexer_test.go

```diff
@@ -1273,74 +1273,58 @@ func startSSHServer(t *testing.T, listener net.Listener) {
 
 func BenchmarkMux_ProxyV2Signature(b *testing.B) {
 	const clusterName = "test-teleport"
-
-	clock := clockwork.NewFakeClockAt(time.Now())
 	tlsProxyCert, caGetter, jwtSigner := getTestCertCAsGetterAndSigner(b, clusterName)
-
-	ca, err := caGetter(context.Background(), types.CertAuthID{
-		Type:       types.HostCA,
-		DomainName: clusterName,
-	}, false)
-	require.NoError(b, err)
-
-	roots := x509.NewCertPool()
-	ok := roots.AppendCertsFromPEM(ca.GetTrustedTLSKeyPairs()[0].Cert)
-	require.True(b, ok)
-
 	ip := "1.2.3.4"
 	sAddr := net.TCPAddr{IP: net.ParseIP(ip), Port: 444}
 	dAddr := net.TCPAddr{IP: net.ParseIP(ip), Port: 555}
+	listener4, err := net.Listen("tcp", "127.0.0.1:")
+	require.NoError(b, err)
+
+	startServing := func(muxListener net.Listener, cluster string) (*Mux, *httptest.Server) {
```

Comment:
This is basically the same as in the unit test: https://github.com/gravitational/teleport/blob/9f08fc37b5dd686d033467b71c443ef4664c9fdf/lib/multiplexer/multiplexer_test.go#L835-L861

The only difference I can see is here we're not using TLS listener - is there a reason, can we unify, factor out and reuse this in both tests? Then both tests become more compact.

Reply:
Ah yeah I can just move that function out. I wanted to keep any refactors to the absolute minimum but considering this is just for setup it shouldn't affect much.